metpy/plots/skewt.py

```diff
@@ -834,10 +834,29 @@ def plot_colormapped(self, u, v, c, bounds=None, colors=None, **kwargs):
             # If we are segmenting by height (a length), interpolate the bounds
             if bounds.dimensionality == {'[length]': 1.0}:
                 bounds = bounds + c[0]  # Make all heights AGL
-                interpolation_heights = concatenate((bounds, c))
+
+                # Find any bounds not in the data and interpolate them
+                interpolation_heights = []
+                for bound in bounds:
```

Comment:
```python
interpolation_heights = [bound.m for bound in bounds if bound not in c]
```
I personally like the list comprehension over the set of `for`, `if`, and creating and appending to the list.

Reply:
Fewer lines = 👍